apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -415,6 +416,159 @@ public partial class BigtableClient
+
+            return await response.AsAsyncEnumerable().SingleOrDefault().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Streams back the contents of all requested rows in key order, optionally
+        /// applying the same Reader filter to each.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This method simply delegates to <see cref="ReadRows(ReadRowsRequest, CallSettings)"/>.
+        /// </para>
+        /// </remarks>
+        /// <param name="tableName">
+        /// The unique name of the table from which to read. Must not be null.
+        /// </param>
+        /// <param name="rows">
+        /// The row keys and/or ranges to read. If null, reads from all rows.
+        /// </param>
+        /// <param name="filter">
+        /// The filter to apply to the contents of the specified row(s). If null,
+        /// reads the entirety of each row.
+        /// </param>
+        /// <param name="rowsLimit">
+        /// The read will terminate after committing to N rows' worth of results.
+        /// If null or 0, returns all results.
+        /// </param>
+        /// <param name="callSettings">
+        /// If not null, applies overrides to this RPC call.
+        /// </param>
+        /// <returns>
+        /// The server stream.
+        /// </returns>
+        public virtual ReadRowsStream ReadRows(
+            TableName tableName,
+            RowSet rows = null,
+            RowFilter filter = null,
+            long? rowsLimit = null,
+            CallSettings callSettings = null) =>
+            ReadRows(
+                new ReadRowsRequest
+                {
+                    TableNameAsTableName = GaxPreconditions.CheckNotNull(tableName, nameof(tableName)),
+                    Rows = rows,
+                    Filter = filter,
+                    RowsLimit = GaxPreconditions.CheckArgumentRange(rowsLimit ?? 0, nameof(rowsLimit), 0, long.MaxValue)
+                },
+                callSettings);
+
+        public partial class ReadRowsStream
+        {
+            private RowAsyncEnumerable _enumerator;
+
+            /// <summary>
+            /// Returns an asynchronous sequence of rows from this set of results.
+            /// </summary>
+            /// <returns>An asynchronous sequence of rows from this set of results.</returns>
+            public IAsyncEnumerable<Row> AsAsyncEnumerable() =>
+                _enumerator ?? (_enumerator = new RowAsyncEnumerable(this));
+
+            private class RowAsyncEnumerable : IAsyncEnumerable<Row>
+            {
+                private bool _enumeratorCreated;
+                private ReadRowsStream _stream;
+
+                public RowAsyncEnumerable(ReadRowsStream stream)
+                {
+                    _stream = stream;
+                }
+
+                public IAsyncEnumerator<Row> GetEnumerator()
+                {
+                    if (_enumeratorCreated)
```

Comment:
Hmm, apparently there's no `CompareExchange` overload that works with bool. Seems like an oversight. Guess this will have to be an `_enumeratorCount` field instead.